android-mvvm/src/main/java/com/example/android_mvvm/ReadOnlyField.java

```diff
@@ -0,0 +1,48 @@
+package com.example.android_mvvm;
+
+import android.databinding.ObservableField;
+import android.support.annotation.NonNull;
+
+import java.util.HashMap;
+
+import rx.Observable;
+import rx.Subscription;
+import rx.functions.Action1;
+
+public class ReadOnlyField<T> extends ObservableField<T> {
+    final Observable<T> source;
+    final HashMap<OnPropertyChangedCallback, Subscription> subscriptions = new HashMap<>();
+
+    public ReadOnlyField(@NonNull Observable<T> source) {
+        this.source = source.share();
+    }
+
+    /**
+     * Calling set method of ReadOnlyField can result in weird behaviour
+     * TODO: Document why set() shouldn't be used
+     */
+    @Override
+    public void set(T value) {
+        super.set(value);
+    }
+
+    @Override
+    public synchronized void addOnPropertyChangedCallback(OnPropertyChangedCallback callback) {
+        super.addOnPropertyChangedCallback(callback);
+        subscriptions.put(callback, source.subscribe(new Action1<T>() {
```

Comment:
can we limit the subscription to 1? I know the same observable is used for all subscribers but still, same could be done like subscribe first time only and unsubscribe when last callback is removed.

About Test still getting hang of it, will update once I know it a little